s4e-backend/src/main/java/pl/cyfronet/s4e/controller/validation/GroupNameValid.java

```diff
@@ -0,0 +1,17 @@
+package pl.cyfronet.s4e.controller.validation;
+
+import javax.validation.Constraint;
+import javax.validation.Payload;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target({ElementType.METHOD, ElementType.FIELD})
+@Retention(RetentionPolicy.RUNTIME)
+@Constraint(validatedBy = GroupNameValidator.class)
+public @interface GroupNameValid {
+    String message() default "{Group.name.not.valid.message}";
```

Comment:
I have a problem with getting this work. When I send an invalid name, the response doesn't have this interpolated.

I added a simple test in GroupControllerTest to check for this:
```java
    @Test
    @WithMockUser
    public void shouldNotCreateGroupWithoutMembersIfNameInvalid() throws Exception {
        CreateGroupRequest groupRequest = CreateGroupRequest.builder()
                .name("__default__")
                .build();

        mockMvc.perform(post(API_PREFIX_V1 + "/institutions/{institution}/groups", slugInstitution)
                .contentType(MediaType.APPLICATION_JSON)
                .content(objectMapper.writeValueAsBytes(groupRequest)))
                .andExpect(status().isBadRequest())
                .andExpect(jsonPath("$.name[0]", containsString("nie może zawierać podkreśleń")));
    }
```

I'm not sure about this method of validation anymore. However, I've stumbled upon this: https://docs.spring.io/spring/docs/current/spring-framework-reference/core.html#validator, it might be just what we are looking for.